src/ray/raylet/worker_pool.h

```diff
@@ -78,6 +80,44 @@ using PopWorkerCallback =
                        PopWorkerStatus status,
                        const std::string &runtime_env_setup_error_message)>;
 
+struct PopWorkerRequest {
+  const rpc::Language language;
+  const rpc::WorkerType worker_type;
+  const JobID job_id;                    // can be Nil
+  const ActorID root_detached_actor_id;  // can be Nil
+  const std::optional<bool> is_gpu;
+  const std::optional<bool> is_actor_worker;
+  const rpc::RuntimeEnvInfo runtime_env_info;
+  const int runtime_env_hash;
+  const std::vector<std::string> dynamic_options;
```

Comment:
avoid const-ing member variables for un-movability, and removal of noexcept move constructor